### ISSUE TITLE
Missing EGL ICD configuration file

### ISSUE THREAD

**Issue by velovix** [NVIDIA/nvidia-docker#1520]
When using nvidia-docker's support for EGL, the runtime does not introduce an [ICD configuration file](https://gitlab.freedesktop.org/glvnd/libglvnd/-/blob/master/src/EGL/icd_enumeration.md) into the environment. This prevents libglvnd from finding the Nvidia EGL module, which causes initialization of EGL to fail.

I'm not the first one to find this:

- #337 was closed because EGL was not yet supported at that time
- #1350 was closed by the author because they found a workaround

The workaround is to introduce the ICD configuration file manually at `/etc/glvnd/egl_vendor.d/10_nvidia.json`.

```json
{
    "file_format_version" : "1.0.0",
    "ICD" : {
        "library_path" : "libEGL_nvidia.so.0"
    }
}
```

The workaround isn't difficult, but considering this file is included in Nvidia's driver packages and is necessary for EGL to function in the way Nvidia intends it to function (through libglvnd), this seems like it should be included.

I've omitted some of the normal debugging information because I think this is nvidia-docker working as designed. I think, though, that this should be reconsidered. If this is not the expected behavior, please let me know and I'll provide some additional information about my environment.

**Comment by elezar** [comment]
Thanks for raising this again @velovix. I have created an internal ticket to track this.

**Comment by bsirang** [comment]
Bumping this issue since I ran into this issue today and discovered the missing json file after combing through system call traces.

**Comment by elezar** [comment]
@bsirang as discussed in https://github.com/NVIDIA/nvidia-container-toolkit/issues/16 we are in the process of improving how these files are handled by the NVIDIA Container Runtime. Would you be able to test the `v1.12.0-rc.1` version from our experimental repository to see if ti addresses your issue?

There is also work in progress to ensure that the DRM devices associated with the selected NVIDIA GPUs are also injected into the container. This will be included in `v1.12.0-rc.2` once we release that.

**Comment by elezar** [comment]
@bsirang we have released `v1.12.0-rc.3` that should include the DRM devices. Would youj be able to check this from your side?

**Comment by elezar** [comment]
@bsirang @velovix we have released `v1.12.0` of the NVIDIA Container Toolkit. This will properly inject Vulkan and EGL ICD files when the NVIDIA Container Runtime is used.

Please test the release and reopen this issue or create a new issue under NVIDIA/nvidia-container-toolkit if you're still having problems or missing functionality.
